python/lsst/pipe/tasks/characterizeImage.py

```diff
@@ -0,0 +1,493 @@
+#
+# LSST Data Management System
+# Copyright 2008-2015 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+import numpy as np
+
+from lsstDebug import getDebugFrame
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.afw.math import BackgroundList
+from lsst.afw.table import SourceTable, SourceCatalog
+from lsst.meas.algorithms import estimateBackground
+from lsst.meas.algorithms.installGaussianPsf import InstallGaussianPsfTask
+from lsst.meas.astrom import AstrometryTask, displayAstrometry
+from .detectAndMeasure import DetectAndMeasureTask
+from .measurePsf import MeasurePsfTask
+from .repair import RepairTask
+
+class CharacterizeImageConfig(pexConfig.Config):
+    """!Config for CharacterizeImageTask"""
+    doMeasurePsf = pexConfig.Field(
+        dtype = bool,
+        default = True,
+        doc = "Measure PSF? If False then keep the existing PSF model (which must exist) "
+            "and use that model for all operations."
+    )
+    doWrite = pexConfig.Field(
+        dtype = bool,
+        default = True,
+        doc = "Persist results?",
+    )
+    doWriteExposure = pexConfig.Field(
+        dtype = bool,
+        default = True,
+        doc = "Write icExp and icExpBackground in addition to icSrc? Ignored if doWrite False.",
+    )
+    psfIterations = pexConfig.RangeField(
+        dtype = int,
+        default = 2,
+        min = 1,
+        doc = "Number of iterations of detect sources, measure sources, estimate PSF. "
+            "If useSimplePsf='all_iter' then 2 should be plenty; otherwise more may be wanted.",
```

Comment:
`useSimplePsf` is a straight boolean, and its docs doesn't mention this possibility.